### ISSUE TITLE
docs/howto: Causing one workflow task to run after another

### ISSUE THREAD

**Issue by jpluscplusm** [cue-lang/docs-and-content#124]
In #123 we track documentation of how to use the output of one task inside another task. In that guide we might well mention that referring to an incomplete field in a task causes a dependency to be created, but we should probably include a page that explains both ways of achieving this:

- `$after: <task>`
- referring to an incomplete field

We should explain their differences in both operation and use case.

Title: it could be the title proposed here, or one such as "Create an explicit dependency between two tasks in a workflow command", or similar. Rather than optimising for the 100% most technically accurate title, let's target the **unaware** user who's simply searching for this functionality: what title would *best* prompt them to read this doc?

**Comment by myitcv** [comment]
Just shared the most basic example of this:

```
package x

import "tool/cli"

command: deps: {
	before: cli.Print & {
		text: "before"
	}
	after: cli.Print & {
		$after: before
		text:   "after"
	}
}
```
